sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -369,6 +372,26 @@ class QueryExecution(
     Utils.redact(sparkSession.sessionState.conf.stringRedactionPattern, message)
   }
 
+  def extendedExplainInfo(append: String => Unit, plan: SparkPlan): Unit = {
+    try {
+      val generator = sparkSession.sparkContext.conf.get(EXTENDED_EXPLAIN_PROVIDER.key)
+      try {
+        val extensionClass = Utils.classForName(generator)
+        val extension = extensionClass.getConstructor().newInstance()
+          .asInstanceOf[ExtendedExplainGenerator]
```

Comment:
Let's call `Utils.loadExtensions`

Reply:
Good suggestion. I was not aware of this method. It is much cleaner to use.
Ironically, we do not use this method to load session extensions!